sdk/ai/ai-projects/review/ai-projects.api.md

```diff
@@ -0,0 +1,1988 @@
+    updateAgent: (assistantId: string, options: UpdateAgentOptions, requestParams?: OptionalRequestParameters) => Promise<AgentOutput>;
+    updateMessage: (threadId: string, messageId: string, options?: UpdateMessageOptions, requestParams?: OptionalRequestParameters) => Promise<ThreadMessageOutput>;
+    updateRun: (threadId: string, runId: string, options?: UpdateRunOptions, requestParams?: OptionalRequestParameters) => Promise<ThreadRunOutput>;
+    updateThread: (threadId: string, options?: UpdateAgentThreadOptions, requestParams?: OptionalRequestParameters) => Promise<AgentThreadOutput>;
+    uploadFile: (data: ReadableStream | NodeJS.ReadableStream, purpose: FilePurpose, fileName?: string, requestParams?: OptionalRequestParameters) => Promise<OpenAIFileOutput>;
+    uploadFileAndPoll: (data: ReadableStream | NodeJS.ReadableStream, purpose: FilePurpose, fileName?: string, pollingOptions?: PollingOptions, requestParams?: OptionalRequestParameters) => Promise<OpenAIFileOutput>;
+}
+
+// @public
+export type AgentStreamEventType = ThreadStreamEvent | RunStreamEvent | RunStepStreamEvent | MessageStreamEvent | ErrorEvent | DoneEvent;
+
+// @public
+export interface AgentThreadCreationOptions {
+    messages?: Array<ThreadMessageOptions>;
+    metadata?: Record<string, string> | null;
+    tool_resources?: ToolResources | null;
+}
+
+// @public
+export interface AgentThreadOutput {
+    created_at: number;
+    id: string;
+    metadata: Record<string, string> | null;
+    object: "thread";
+    tool_resources: ToolResourcesOutput | null;
+}
+
+// @public (undocumented)
+export class AIProjectsClient {
+    constructor(endpointParam: string, subscriptionId: string, resourceGroupName: string, projectName: string, credential: TokenCredential, options?: AIProjectsClientOptions);
+    readonly agents: AgentsOperations;
+    readonly connections: ConnectionsOperations;
+    static fromConnectionString(connectionString: string, credential: TokenCredential, options?: AIProjectsClientOptions): AIProjectsClient;
+    readonly telemetry: TelemetryOperations;
+}
+
+// @public (undocumented)
+export interface AIProjectsClientOptions extends ProjectsClientOptions {
+}
+
+// @public
+export type ApiResponseFormat = string;
+
+// @public
+export type ApiResponseFormatOutput = string;
+
+// @public
+export interface AppInsightsPropertiesOutput {
+    ConnectionString: string;
+}
+
+// @public
+export interface ApplicationInsightsConfiguration extends InputDataParent {
+    connectionString?: string;
+    query: string;
+    resourceId: string;
+    serviceName: string;
+}
+
+// @public
+export interface ApplicationInsightsConfigurationOutput extends InputDataOutputParent {
+    connectionString?: string;
+    query: string;
+    resourceId: string;
+    serviceName: string;
+    // (undocumented)
+    readonly type: "app_insights";
```

Comment:
We use the keyword `kind` for these sorts of things, not `type`

Reply:
Auto generated from typespec, How can we make this specific to JS ?

Reply:
I'd recommend using `kind` in the TypeSpec as well. If not, there are customization options that you can add to the `client.tsp` specific to JS naming. You could check with the TypeSpec team if you need help with that